### PR TITLE
chore(deps): update to Moby v26.1 & buildx v0.14

### PR DESCRIPTION
**Pull request by milas**
* `moby/buildkit` commit comes from `docker/buildx`
* v26.1-dev from `moby/moby` & `docker/cli`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,18 +4,18 @@ go 1.21
...
+	github.com/containerd/containerd v1.7.15
 	github.com/davecgh/go-spew v1.1.1
 	github.com/distribution/reference v0.6.0
-	github.com/docker/buildx v0.13.1
-	github.com/docker/cli v26.0.1+incompatible
+	github.com/docker/buildx v0.14.0
+	github.com/docker/cli v26.0.1-0.20240419172455-118d6bafe06e+incompatible
 	github.com/docker/cli-docs-tool v0.7.0
-	github.com/docker/docker v26.0.1+incompatible
+	github.com/docker/docker v26.0.1-0.20240419122302-ee8b788538ea+incompatible
```
[the remark this answers is not in the document]
Comment (vvoland) [reply go.mod:18]: Yeah, the engine shouldn't change now. But with CLI we still want to merge https://github.com/docker/cli/pull/5030 on Monday.

[the remark this answers is not in the document]
Comment (milas) [reply go.mod:18]: OK, that commit shouldn't impact Compose's usage of `docker/cli` as a library, so we should be fine here (Also, it doesn't have any hooks currently anyway 😛)

[the remark this answers is not in the document]
Comment (vvoland) [reply go.mod:18]: You can use v26.1.0 tags now 😄

**Review by glours — APPROVED** [review]
LGTM

**Review by ndeloof — APPROVED** [review]

**Pull request merged**